### PR TITLE
Resolve tweaks

### PR DESCRIPTION
**Pull request by nnethercote**
A couple of small perf improvements, and some minor refactorings, all in `rustc_resolve`.

r? @petrochenkov

### CODE REVIEW

**Comment by nnethercote** [comment]
@petrochenkov: I was going to request a different reviewer because (a) I know you are busy, and (b) none of these changes involve any resolver-specific expertise. But you can review if you want.

**Comment by jieyouxu** [comment]
I can take a look.

r? jieyouxu

File: compiler/rustc_resolve/src/ident.rs [hunk 1]
```diff
@@ -814,34 +793,10 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
...
-            ignore_import,
-        )
-        .map_err(|(determinacy, _)| determinacy)
-    }
-
     /// Attempts to resolve `ident` in namespaces `ns` of `module`.
     /// Invariant: if `finalize` is `Some`, expansion and import resolution must be complete.
     #[instrument(level = "debug", skip(self))]
-    fn resolve_ident_in_module_unadjusted_ext(
+    fn resolve_ident_in_module_unadjusted(
```
Comment (jieyouxu) [inline compiler/rustc_resolve/src/ident.rs:799]: Suggestion: github UI doesn't let me comment outside the diff, but should we introduce an enum for `RestrictedShadowing`? The `false` looks very uninformative and it is not documented.

**Review by jieyouxu — APPROVED** [review]
Thanks, one tiny nit regarding a `bool` param for a fn that has multiple params that looks confusing at call-sites. But otherwise, this looks like a nice cleanup!

**Comment by petrochenkov** [comment]
>But you can review if you want.

Yes, I self assign if I want to review, or at least take a look.

**Comment by jieyouxu** [comment]
(My bad I didn't notice it)

File: compiler/rustc_resolve/src/lib.rs [hunk 2]
```diff
@@ -534,8 +534,8 @@ struct BindingKey {
     /// identifier.
     ident: Ident,
     ns: Namespace,
-    /// 0 if ident is not `_`, otherwise a value that's unique to the specific
-    /// `_` in the expanded AST that introduced this binding.
+    /// 0 if ident is not `_` or ``, otherwise a value that's unique to the specific
+    /// `_`/`` in the expanded AST that introduced this binding.
```
Comment (petrochenkov) [inline compiler/rustc_resolve/src/lib.rs:538]: https://github.com/rust-lang/rust/pull/115367 was reverted in https://github.com/rust-lang/rust/pull/131045, and even if it weren't `kw::Empty` shouldn't have been used for this anyway.

File: compiler/rustc_resolve/src/ident.rs (hunk 1 above)
Comment (nnethercote) [reply compiler/rustc_resolve/src/ident.rs:799]: I'll do an enum.

File: compiler/rustc_resolve/src/lib.rs (hunk 2 above)
Comment (nnethercote) [reply compiler/rustc_resolve/src/lib.rs]: Huh, interesting. `empty_disambiguator` was added in #115367 but wasn't removed by the revert in #131045. @compiler-errors, you did the revert, do you remember how you went about it -- was it a straightforward reversion? Was leaving `empty_disambiguator` behind an oversight?

There is one test (`tests/ui/dollar-crate/dollar-crate-is-keyword.rs`) that hits the empty identifier path. But if I remove `empty_disambiguator` all tests pass. So I've removed the commit changing this comment and added a new commit that removes `empty_disambiguator`.

**Comment by nnethercote** [comment]
I have updated. I removed one old commit (the one updating the comment on `BindingKey::disambiguator`) and added two new commits (one adding the `Shadowing` enum, and one removing `Resolver::empty_disambiguator`).

**Pull request merged**